### ISSUE TITLE
crc setup error "Failed to start libvirt 'crc' network"

### ISSUE THREAD

**Issue by adrianbele** [code-ready/crc#1812]
### General information

  * OS: CentOS Linux release 8.3.2011
  * Hypervisor: Hyper-V 
  * Did you run `crc setup` before starting it (Yes/No)? yes

## CRC version
```bash
# Put the output of `crc version`
CodeReady Containers version: 1.20.0+ef3f80d
OpenShift version: 4.6.6 (embedded in executable)
```
  
## CRC status
```bash
CRC VM:          Running
OpenShift:       Not Reachable (v4.6.6)
Disk Usage:      15.03GB of 32.72GB (Inside the CRC VM)
Cache Usage:     13.69GB
Cache Directory: /home/redscorpion/.crc/cache
```

## CRC config
```bash
#empty
```

## Host Operating System
```bash
NAME="CentOS Linux"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
```

### Steps to reproduce

  1. crc setup 

### Expected
No error

### Actual
End with error
```bash
...
INFO Checking if libvirt 'crc' network is active
INFO Starting libvirt 'crc' network
Failed to start libvirt 'crc' network
 exit status 1: error: Failed to start network crc
error: COMMAND_FAILED: '/usr/sbin/iptables -w10 -w --table filter --insert LIBVIRT_INP --in-interface crc --protocol tcp --destination-port 67 --jump ACCEPT' failed: iptables: No chain/target/match by that name.
```
### Fix
This fixed the issue for me

```bash
$ crc delete -f 
$ crc cleanup
$ sudo iptables -t filter -F
$ sudo iptables -t filter -X
$ crc setup
#no error this time
 ```
### Logs

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by praveenkumar** [comment]
This is duplicate of https://github.com/code-ready/crc/issues/494 one.

**Comment by adrianbele** [comment]
#494 solution works too 
The troubleshooting section in the documents need an update

**Comment by praveenkumar** [comment]
@adrianbele can you provide steps to reproduce it since last time we were not able to reproduce it and hence it was closed without any update.

**Comment by adrianbele** [comment]
on this the above machine,  192.168.1.25 , I had a failed minishift remote install   
minishift start --vm-driver generic --remote-ipaddress 192.168.1.25
Then start with crc setup 
I did not clean up the minishift install from the 192.168.1.25

**Comment by harshchoudhary727** [comment]
**For Ubuntu**

_Openshift Version_- 4.6.15
_OS-_ Ubuntu 20
_Issue:_ "Failed to start libvirt 'crc' network"


```

 crc delete -f 
crc cleanup
sudo iptables -t filter -F
sudo iptables -t filter -X
```


```
apt remove virt-*

apt install virt-* 

apt-get update -y
```

**after updating reboot the laptop/desktop**



 **`crc setup --log-level debug`**


**ISSUE IS RESOLVED**

![image](https://user-images.githubusercontent.com/66783694/108818619-a7e6a700-75df-11eb-9cb6-be7c34f6ef57.png)

**Comment by cfergeau** [comment]
@adrianbele did you manage to figure out what was wrong? Can this issue be closed?

**Comment by praveenkumar** [comment]
Check https://github.com/kubernetes/minikube/issues/8772#issuecomment-693584896 if you face issue with nftable.

```
Failed to start libvirt 'crc' network 
 exit status 1: error: Failed to start network crc
error: error from service: changeZoneOfInterface: COMMAND_FAILED: 'python-nftables' failed: 
JSON blob:
{"nftables": [{"metainfo": {"json_schema_version": 1}}, {"add": {"rule": {"family": "inet", "table": "firewalld", "chain": "filter_IN_libvirt_allow", "expr": [{"match": {"left": {"payload": {"protocol": "udp", "field": "dport"}}, "op": "==", "right": 67}}, {"match": {"left": {"ct": {"key": "state"}}, "op": "in", "right": {"set": ["new", "untracked"]}}}, {"accept": null}]}}}, {"add": {"rule": {"family": "inet", "table": "firewalld", "chain": "filter_IN_libvirt_allow", "expr": [{"match": {"left": {"payload": {"protocol": "udp", "field": "dport"}}, "op": "==", "right": 547}}, {"match": {"left": {"ct": {"key": "state"}}, "op": "in", "right": {"set": ["new", "untracked"]}}}, {"accept": null}]}}}, {"add": {"rule": {"family": "inet", "table": "firewalld", "chain": "filter_IN_libvirt_allow", "expr": [{"match": {"left": {"payload": {"protocol": "tcp", "field": "dport"}}, "op": "==", "right": 53}}, {"match": {"left": {"ct": {"key": "state"}}

```

**Comment by cfergeau** [comment]
No more input since March 19th, and my understanding is that the issue has been successfully fixed or worked around, closing.

**Comment by hbelmiro** [comment]
@cfergeau @praveenkumar I'm facing this very same issue with 2.10.2 on Fedora 36 and Docker installed.

**Comment by tomaszstachera** [comment]
> **For Ubuntu**
> 
> _Openshift Version_- 4.6.15 _OS-_ Ubuntu 20 _Issue:_ "Failed to start libvirt 'crc' network"
> 
> ```
> 
>  crc delete -f 
> crc cleanup
> sudo iptables -t filter -F
> sudo iptables -t filter -X
> ```
> 
> ```
> apt remove virt-*
> 
> apt install virt-* 
> 
> apt-get update -y
> ```
> 
> **after updating reboot the laptop/desktop**
> 
> **`crc setup --log-level debug`**
> 
> **ISSUE IS RESOLVED**
> 
> ![image](https://user-images.githubusercontent.com/66783694/108818619-a7e6a700-75df-11eb-9cb6-be7c34f6ef57.png)

This one is not working in my case - Ubuntu Server 20.04
